Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -1803,6 +1813,33 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
     fillCaloTable(&caloEventHandler, caloPHOSCells, caloPHOSCellsTRGR, caloCellsCursor, caloCellsTRGTableCursor, bcsMap, 0);
   }
 
+  // fill cpvcluster table
+  if (mInputSources[GIndex::CPV]) {
+    float posX, posZ;
+    for (auto& cpvEvent : cpvTrigRecs) {
+      uint64_t bc = cpvEvent.getBCData().toLong();
+      auto item = bcsMap.find(bc);
+      int bcID = -1;
+      if (item != bcsMap.end()) {
+        bcID = item->second;
+      } else {
+        LOG(fatal) << "Error: could not find a corresponding BC ID for a CPV Trigger Record; BC = " << bc;
+      }
+      for (int iClu = cpvEvent.getFirstEntry(); iClu < cpvEvent.getFirstEntry() + cpvEvent.getNumberOfObjects(); iClu++) {
+        auto& clu = cpvClusters[iClu];
+        clu.getLocalPosition(posX, posZ);
+        cpvClustersCursor(0,
+                          bcID,
+                          posX,
+                          posZ,
+                          clu.getEnergy(),
```

Comment:
Please use the truncate float fraction method to reduce to the expected resolution

Reply:
All right! Is there any description of truncateFloatFraction() function?

Reply:
I don't think so, but you can look at the usage in the same file and the function itself is also very short